src/library.js

```diff
@@ -4819,12 +4819,12 @@ LibraryManager.library = {
   llvm_dbg_value: true,
   llvm_debugtrap: true,
   llvm_ctlz_i32: true,
-  emscripten_asm_const: true,
-  emscripten_asm_const_int: true,
-  emscripten_asm_const_double: true,
-  emscripten_asm_const_int_sync_on_main_thread: true,
-  emscripten_asm_const_double_sync_on_main_thread: true,
-  emscripten_asm_const_async_on_main_thread: true,
+  emscripten_asm_const: function() {},
+  emscripten_asm_const_int: function() {},
+  emscripten_asm_const_double: function() {},
+  emscripten_asm_const_int_sync_on_main_thread: function() {},
+  emscripten_asm_const_double_sync_on_main_thread: function() {},
+  emscripten_asm_const_async_on_main_thread: function() {},
```

Comment:
why is this needed, and why just these functions?

Reply:
Without this change the list of functions in `forwarded_json['Functions']['libraryFunctions']` won't include these since (I suppose) it sees them as data rather than functions?

We need the list of include these functions otherwise this new flag won't work with any EM_ASM-using code.

The comment above here says `misc definitions to avoid unnecessary unresolved symbols from fastcomp`.  Which I guess means that fastcomp undefined symbol checker also depends on these, but that is does't care if they are functions of non-functions.   I will update the command and apply this change to all those other ones too.